[host] dxgi: add microsecond-scale sleep capability to d3d12

VMFortress opened the pull request:
The usleep() call lets d3d12 match the sleep calls of d3d11 while
maintaining the current millisecond-scale sleep interface in the
configuration file.

gnif commented:
We opted to not use `usleep` here due to system resolution issues. If you truly need sub ms accuracy please use `nanosleep`.

quantum5 commented on host/platform/Windows/capture/DXGI/src/d3d12.c:
```diff
@@ -120,8 +121,8 @@ static bool d3d12_create(struct DXGIInterface * intf)
 
-  this->copySleep = option_get_int("dxgi", "d3d12CopySleep");
-  DEBUG_INFO("Sleep before copy : %d ms", this->copySleep);
+  this->copySleep = option_get_float("dxgi", "d3d12CopySleep");
```
`d3d12CopySleep` is declared as `OPTION_TYPE_INT`, so this would just generate garbage.

quantum5 commented on host/platform/Windows/capture/DXGI/src/d3d12.c:
```diff
@@ -346,7 +347,7 @@ static bool d3d12_copyFrame(Texture * parent, ID3D11Texture2D * src)
 
   if (this->copySleep > 0)
-    Sleep(this->copySleep);
+    usleep((uint64_t)(this->copySleep * 1000));
```
We have a custom `nsleep` handler that uses `NtDelayExecution` and is supposed to be more accurate.

VMFortress commented:
Updated to `nsleep()`. Apologies as I was using `d3d11.c` as reference but I think we've discussed some of this before.

Also, I fixed the type define in `dxgi.c`.

quantum5 reviewed (approved):
LGTM
